### PR TITLE
Upgrade Elastic APM Go Agent to v1.1.1

### LINKED ISSUE #1694: Update self-instrumentation [Elastic Go APM] agent
v1.1.1 Is the [latest release](https://github.com/elastic/apm-agent-go/releases) right now.  Required for elastic/apm-server#1671

### PR DESCRIPTION
**Pull request by graphaelli**
* lifts single host restriction on `apm-server.instrumentation.hosts`
* updates tests to trigger activity using v2 endpoints, in preparation for #1671 

closes #1694

### CODE REVIEW

**Comment by graphaelli** [comment]
jenkins, retest this please.  Failed per #1624

File: vendor/go.elastic.co/apm/Jenkinsfile [hunk 1]
```diff
@@ -0,0 +1,236 @@
+#!/usr/bin/env groovy
```
Comment (simitt) [inline vendor/go.elastic.co/apm/Jenkinsfile]: Is it expected that this file is vendored?

Comment (simitt) [reply vendor/go.elastic.co/apm/Jenkinsfile]: we can have a look at that later on.

**Comment by graphaelli** [comment]
jenkins, retest this please

**Pull request merged**